test/framework/src/main/java/org/elasticsearch/transport/MockTcpTransport.java

```diff
@@ -294,26 +324,21 @@ public void onFailure(Exception e) {
                 @Override
                 protected void doRun() throws Exception {
                     StreamInput input = new InputStreamStreamInput(new BufferedInputStream(activeChannel.getInputStream()));
-                    while (isOpen.get()) {
+                    while (isOpen.get() && !Thread.currentThread().isInterrupted()) {
```

Comment:
who interrupts this thread that didn't call the close method and thus cancellableThreads.cancel()?